src/scopyExceptionHandler.h

```diff
@@ -0,0 +1,27 @@
+#ifndef SCOPYEXCEPTIONHANDLER_H
+#define SCOPYEXCEPTIONHANDLER_H
+
+#include <QMessageBox>
+#include <QApplication>
+#include <config.h>
+#include <libm2k/m2kexceptions.hpp>
+#include <scopyApplication.hpp>
+
+using namespace adiscope;
+
+#define HANDLE_EXCEPTION(e) \
+	if(GetScopyApplicationInstance()->getDebugMode()) { \
+		QMessageBox msg; \
+		QString str; \
+		auto me = dynamic_cast<libm2k::m2k_exception*>(&e);\
```

Comment:
also if your doing functions instead of `#define` you could have different overloads for different exceptions so you could get rid of the `dynamic_cast`.

Reply:
cannot do define because of __FILE__ and __LINE__ limitations.